app/models/smart_multilingual_tokenizer.rb

```diff
@@ -0,0 +1,35 @@
+class SmartMultilingualTokenizer
+  Response = Data.define(:token, :start_offset, :end_offset, :type)
+  Analyzed = Data.define(:language, :tokens)
+  def initialize
+    @client = Elasticsearch::Client.new(hosts: [ "localhost:9200" ])
+    @index_name = "smart_multilingual"
+  end
+
+  # Analyze text: detect language and tokenize using Elasticsearch analyzer
+  def analyze_multilingual_text(text)
+    Analyzed.new(detect_language(text),
+                 tokenize_with_standard_analyzer(text))
+  end
+
+  private
+
+  # Detect language using CLD3 (refactored to be more idiomatic Ruby)
+  def detect_language(text)
+    lang = CLD3::NNetLanguageIdentifier.new(0, 400).find_language(text)&.language
+    { ja: "ja", ko: "ko", en: "en" }[lang] || "unknown"
+  end
+
+  # Tokenize text using Elasticsearch's analyze API
+  def tokenize_with_standard_analyzer(text)
+    @client.indices.analyze(
+      index: @index_name,
+      body: { analyzer: "standard", text: text }
+    )["tokens"].map do |token|
+      Response.new(token["token"].downcase,
+                  token["start_offset"],
+                  token["end_offset"],
+                  token["type"])
```

Comment:
```suggestion
      Response.new token["token"].downcase,
                   token["start_offset"],
                   token["end_offset"],
                   token["type"]
```

Rubyでは、こんな感じで括弧なしで引数を指定して、引数を縦に並べる書き方をします。

Reply:
https://github.com/jdkim/textae_campus/pull/15/commits/20cbcf8f4f1ba94f5e1c451acca0259eb4786447 で修正しました。